### PR TITLE
Add com.ktechpit.wonderwall

### PR DESCRIPTION
**Pull request by keshavbhatt**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.ktechpit.wonderwall.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+app-id: com.ktechpit.wonderwall
+base: io.qt.qtwebkit.BaseApp
+base-version: '5.12'
+runtime: org.kde.Platform
+runtime-version: '5.12'
+sdk: org.kde.Sdk
+command: wonderwall
+finish-args:
+  - --share=network
```
Comment (keshavbhatt) [inline com.ktechpit.wonderwall.yaml:9]: Required cause the app uses internet to pull wallpapers.

File: com.ktechpit.wonderwall.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+app-id: com.ktechpit.wonderwall
+base: io.qt.qtwebkit.BaseApp
+base-version: '5.12'
+runtime: org.kde.Platform
+runtime-version: '5.12'
```
Comment (hfiguiere) [inline com.ktechpit.wonderwall.yaml:5]: 5.15 I said.

File: com.ktechpit.wonderwall.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+sdk: org.kde.Sdk
+command: wonderwall
+finish-args:
+  - --share=network
+  - --share=ipc
+  - --socket=x11
```
Comment (hfiguiere) [inline com.ktechpit.wonderwall.yaml:11]:
```suggestion
```

File: com.ktechpit.wonderwall.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      - DEFINES+=USE_DBUS
+      - INCLUDEPATH+="${FLATPAK_DEST}/include/QtWebKitWidgets"
+      - INCLUDEPATH+="${FLATPAK_DEST}/include/QtWebKit"
+      - LRELEASE=lrelease
+    build-options:
+      env:
+        QMAKEPATH: "/app/lib"
+    sources:
+      - type: archive
+        url: http://ktechpit.com/USS/WW/dist/wonderwall_7.0.388_flathub.zip
```
Comment (hfiguiere) [inline com.ktechpit.wonderwall.yaml:41]: since you didn't reply on matrix, you know this give anyone access to the source code...

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.ktechpit.wonderwall.yaml (hunk 2 above)
Comment (keshavbhatt) [reply com.ktechpit.wonderwall.yaml:5]: I tried building with Qt5.15 but the build require many changes in source code, which is due to Qt's migration step to version 6. My app was built with Qt5.5.

File: flathub.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "skip-arches": ["arm"]
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply flathub.json:2]: @nedrichards `aarch64` builds fine. But this still built against a runtime that builds `arm`

[the remark this answers is not in the document]
Comment (hfiguiere) [reply flathub.json:2]: nevermind. actually aarch64 should be disabled too as this is just binaries and they are for x86_64. Unless you have aarch64 binaries.

Comment (hfiguiere) [inline flathub.json:2]:
```suggestion
  "skip-arches": ["arm", "aarch64"]
```

File: com.ktechpit.wonderwall.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  
+modules:
+  - name: ww
+    buildsystem: simple
+    build-commands:
+     - unzip wonderwall_7.0.388_flathub.zip -d ${FLATPAK_DEST}
+    sources:
+    - type: file
+      #path: wonderwall_7.0.388_flathub.zip
+      url: https://github.com/keshavbhatt/WonderWall-Packaging/releases/download/v7.0/wonderwall_7.0.388_flathub.zip
```
Comment (hfiguiere) [inline com.ktechpit.wonderwall.yaml:34]:
```suggestion
      url: https://github.com/keshavbhatt/WonderWall-Packaging/releases/download/v7.0/wonderwall_7.0.388_flathub.zip
      only-arches:
        - x86_64
```

**Comment by keshavbhatt** [comment]
bot, build com.ktechpit.wonderwall

File: flathub.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (nedrichards) [reply flathub.json]: Ah yes, I'd forgotten that this was 5.14. So yes, both arm architectures should be disabled.

**Pull request closed without merging**